Library/Formula/dcmtk.rb

```diff
@@ -12,39 +32,32 @@ class Dcmtk < Formula
     sha1 "044eeba49ae368d5f8cce753d54dcaa9b8d18fbb" => :lion
   end
 
-  option 'with-docs', 'Install development libraries/headers and HTML docs'
-  option 'with-openssl', 'Configure DCMTK with support for OpenSSL'
+  option "with-docs", "Install development libraries/headers and HTML docs"
+  option "with-openssl", "Configure DCMTK with support for OpenSSL"
+  option "with-libiconv", "Build with libiconv"
 
-  depends_on 'cmake' => :build
+  depends_on "cmake" => :build
   depends_on "libpng"
-  depends_on 'libtiff'
+  depends_on "libtiff"
   depends_on "doxygen" => :build if build.with? "docs"
-
-  # This roughly corresponds to thefollowing upstream patch:
-  #
-  #   http://git.dcmtk.org/web?p=dcmtk.git;a=commitdiff;h=dbadc0d8f3760f65504406c8b2cb8633f868a258
-  #
-  # However, this patch can't be applied as-is, since it refers to
-  # some files that don't exist in the 3.6.0 release.
-  #
-  # This patch can be dropped once DCMTK makes a new release, but
-  # since this is a very rare occurrence (the last development preview
-  # release is from mid 2012), it seems justifiable to keep the patch
-  # ourselves for a while.
-  patch :DATA
+  depends_on "openssl" => :optional
+  depends_on "homebrew/dupes/libiconv" => :optional
 
   def install
     ENV.m64 if MacOS.prefer_64_bit?
 
     args = std_cmake_args
-    args << '-DDCMTK_WITH_DOXYGEN=YES' if build.with? "docs"
-    args << '-DDCMTK_WITH_OPENSSL=YES' if build.with? "openssl"
-    args << '..'
+    args << "-DDCMTK_WITH_DOXYGEN=YES" if build.with? "docs"
+    args << "-DDCMTK_WITH_OPENSSL=YES" if build.with? "openssl"
+    args << "-DWITH_OPENSSLINC=#{Formula["openssl"].opt_prefix}" if build.with? "openssl"
+    args << "-DDCMTK_WITH_ICONV=YES" if build.with? "libiconv"
+    args << "-DLIBICONV_DIR=#{Formula["libiconv"].opt_prefix}" if build.with? "libiconv"
+    args << ".."
 
-    mkdir 'build' do
-      system 'cmake', *args
-      system 'make DOXYGEN' if build.with? "docs"
-      system 'make install'
+    mkdir "build" do
+      system "cmake", *args
+      system "make", "DOXYGEN" if build.with? "docs"
+      system "make", "install"
     end
   end
```

Comment:
Can a test be added to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!